pkgs/development/libraries/libarchive/default.nix

```diff
@@ -22,43 +22,47 @@
 , cmake
 , nix
 , samba
+, buildPackages
 }:
 
+let
+  autoreconfHook = buildPackages.autoreconfHook269;
```

Comment:
I don't think we need buildPackages at all when using autoreconfHook269 in nativeBuildInputs. It was just in top-level to not break due to the override.